rusoto/signature/src/signature.rs

```diff
@@ -134,11 +175,39 @@ impl SignedRequest {
         self.payload = payload.map(|chunk| SignedRequestPayload::Buffer(chunk.into()));
     }
 
-    /// Sets the new body (payload) as a stream
+    #[cfg(feature = "futures-01")]
+    /// Sets the new body (payload)
     pub fn set_payload_stream(&mut self, stream: ByteStream) {
         self.payload = Some(SignedRequestPayload::Stream(stream));
     }
 
+    #[cfg(feature = "async-await")]
+    /// Sets the new body (payload)
+    pub fn set_payload_stream(
+        &mut self,
+        payload: Pin<Box<dyn Stream<Item = Result<Bytes, std::io::Error>> + Send + Sync + 'static>>,
+    ) {
+        self.payload = Some(SignedRequestPayload::Stream(payload))
+    }
+
+    #[cfg(feature = "futures-01")]
+    /// Computes and sets the Content-MD5 header based on the current payload.
+    ///
+    /// Has no effect if the payload is not set, or is not a buffer.
+    pub fn set_content_md5_header(&mut self) {
+        let digest;
+        if let Some(SignedRequestPayload::Buffer(ref payload)) = self.payload {
+            digest = Some(md5::compute(payload));
+        } else {
+            digest = None;
+        }
+        if let Some(digest) = digest {
+            // need to deref digest and then pass that reference:
+            self.add_header("Content-MD5", &base64::encode(&(*digest)));
```

Comment:
It looks like you could do away with the digest bar and hoist line 206 into 200. Less code and easier to follow